src/Http/Routing/src/EndpointMiddleware.cs

```diff
@@ -62,10 +63,13 @@ public Task Invoke(HttpContext httpContext)
                 }
                 catch (Exception exception)
                 {
+                    EndpointMiddlewareEventSource.Log.ExecutedEndpoint(endpoint.DisplayName);
+                    EndpointMiddlewareEventSource.Log.FailedEndpoint(endpoint.DisplayName);
```

Comment:
May it be "better" to handle the _executed_ within the _failed_?
So that here is just one method call, and in the internals of `Log.FailedEndpoint` this case gets handled and has the potential to adapt it later if needed.